Content.Server/GameObjects/Components/Tag/TagComponent.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+
+namespace Content.Server.GameObjects.Components.Tag
+{
+    [RegisterComponent]
+    public class TagComponent : Component
+    {
+        public override string Name => "Tag";
+
+        private HashSet<string> _tags = new(0);
```

Comment:
I think using a 0 capacity is technically slower if the default capacity is already 0 and it doesn't save memory. pjb told me this about lists and I just checked and I'm pretty sure it applies to hashsets too.

Reply:
Fixed.